processors/baggage/baggagetrace/processor.go

```diff
@@ -11,23 +11,36 @@ import (
 	"go.opentelemetry.io/otel/sdk/trace"
 )
 
+// BaggageKeyPredicate is a function that returns true if the baggage key should be added to the span.
+type BaggageKeyPredicate func(baggageKey string) bool
+
+// AllowAllBaggageKeys is a BaggageKeyPredicate that allows all baggage keys.
+var AllowAllBaggageKeys BaggageKeyPredicate = func(string) bool { return true }
+
 // SpanProcessor is a processing pipeline for spans in the trace signal.
-type SpanProcessor struct{}
+type SpanProcessor struct {
+	baggageKeyPredicate BaggageKeyPredicate
+}
 
 var _ trace.SpanProcessor = (*SpanProcessor)(nil)
 
 // New returns a new SpanProcessor.
 //
 // The Baggage span processor duplicates onto a span the attributes found
 // in Baggage in the parent context at the moment the span is started.
-func New() trace.SpanProcessor {
-	return &SpanProcessor{}
+// The predicate function is used to filter which baggage keys are added to the span.
+func New(baggageKeyPredicate BaggageKeyPredicate) trace.SpanProcessor {
```

Comment:
Please use the options pattern so that the new configuration option is optional https://github.com/open-telemetry/opentelemetry-go/blob/main/CONTRIBUTING.md#configuration

Reply:
I've intentionally used a constructor parameter to make the predicate function required. If it's an optional parameter it's easy for a user to miss and not set an appropriate filter. This helps it be safe by default and offer a connivence allow all filter.

Reply:
How would you add a new configuration without introducing a breaking change?

I do not think it is convenient to require a filter. 

> offer a connivence allow all filter.

Users may miss it as well.

Reply:
Agree, this is a breaking change but I think it's worth it.

Requiring a filter isn't convenient but is safer than presuming all baggage keys should be copied, which is the current behaviour.

It's possible users miss the convenience function `baggagetrace.AllowAllBaggageKeys`, but they won't be able to add the processor without providing a filter function so they at least have to consider what should be allowed.

Hopefully the documentation is extensive enough they discover it and use it.

Reply:
Resolving also per https://github.com/open-telemetry/opentelemetry-go-contrib/pull/5619#pullrequestreview-2094879032